main.go

```diff
@@ -29,17 +29,7 @@ func Handler(ctx context.Context, request events.APIGatewayProxyRequest) (string
 		"DELETE DATA": true,
 	}
 
-	log.Printf("API GATEWAY REQUEST: %+v", request)
-
-	// The body that we get from API gateway is decoded, so we need to re-encode it
-	// However, we need to first remove the query/upadte directive before encoding.
-	bodyIn := request.Body
-	i := strings.Index(bodyIn, "=")
-	queryPrefix := bodyIn[:i]
-	bodyIn = bodyIn[i+1:]
-	bodyIn = fmt.Sprintf("%s=%s", queryPrefix, url.QueryEscape(bodyIn))
-
-	proxyReq, err := http.NewRequest("POST", os.Getenv("RIALTO_SPARQL_ENDPOINT"), strings.NewReader(bodyIn))
+	proxyReq, err := http.NewRequest("POST", os.Getenv("RIALTO_SPARQL_ENDPOINT"), strings.NewReader(encodeBody(request.Body)))
```

Comment:
Could encode body return a reader?

Reply:
Good catch, that makes more sense. Done.